From 3604664a6952f9b1526722b0c81ede70da656ede Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Fri, 13 May 2022 10:40:34 +0300
Subject: [PATCH 02/30] ddr: err050543: Sync dtb

By adding 'ddr_errata_reserved' entry we prevent any
writings into the reserved memory buffer by U-Boot.

Issue: ALB-8823
Issue: ALB-8781
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/s32cc.dtsi  |  6 ++++++
 arch/arm/dts/s32g2.dtsi  | 14 ++++++++++++++
 arch/arm/dts/s32r45.dtsi | 14 ++++++++++++++
 3 files changed, 34 insertions(+)

diff --git a/arch/arm/dts/s32cc.dtsi b/arch/arm/dts/s32cc.dtsi
index 05657927cc..9acf8a3c19 100644
--- a/arch/arm/dts/s32cc.dtsi
+++ b/arch/arm/dts/s32cc.dtsi
@@ -394,6 +394,12 @@
 			};
 		};
 
+		ddr_errata: ddr_errata@403C0000 {
+			compatible = "nxp,s32cc-ddr";
+			reg = <0x0 0x403C0000 0x0 0x100>;
+			status = "disabled";
+		};
+
 		pcie0: pcie@40400000 {
 			compatible = "nxp,s32cc-pcie";
 			reg = <0x00 0x40400000 0x0 0x00001000   /* dbi registers */
diff --git a/arch/arm/dts/s32g2.dtsi b/arch/arm/dts/s32g2.dtsi
index 925ec312ee..297f985432 100644
--- a/arch/arm/dts/s32g2.dtsi
+++ b/arch/arm/dts/s32g2.dtsi
@@ -8,6 +8,7 @@
  * (at your option) any later version.
  */
 
+#include <dt-bindings/ddr-errata/s32-ddr-errata.h>
 #include "s32g.dtsi"
 / {
 	model = "NXP S32G2XX";
@@ -22,6 +23,14 @@
 		device_type = "memory";
 		reg = <0x8 0x80000000 0 0x80000000>;
 	};
+
+	reserved-memory {
+		ddr_errata_reserved: ddr@85000000 {
+			reg = <0x0 DDR_ERRATA_REGION_BASE
+				0x0 DDR_ERRATA_REGION_SIZE>; /* 4 KB */
+			no-map;
+		};
+	};
 };
 
 &cmu {
@@ -35,3 +44,8 @@
 &siul2_1_nvram {
 	compatible = "nxp,s32g2-siul2_1-nvram";
 };
+
+&ddr_errata {
+	memory-region = <&ddr_errata_reserved>;
+	status = "okay";
+};
diff --git a/arch/arm/dts/s32r45.dtsi b/arch/arm/dts/s32r45.dtsi
index 1d871f3e54..8da95a67f4 100644
--- a/arch/arm/dts/s32r45.dtsi
+++ b/arch/arm/dts/s32r45.dtsi
@@ -8,6 +8,7 @@
  * (at your option) any later version.
  */
 #include <dt-bindings/clock/s32r45-scmi-clock.h>
+#include <dt-bindings/ddr-errata/s32-ddr-errata.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/nvmem/s32cc-siul2-nvmem.h>
 #include <dt-bindings/pinctrl/s32r45-pinctrl.h>
@@ -32,6 +33,14 @@
 		reg = <0x8 0x80000000 0 0x80000000>;
 	};
 
+	reserved-memory {
+		ddr_errata_reserved: ddr@85000000 {
+			reg = <0x0 DDR_ERRATA_REGION_BASE
+				0x0 DDR_ERRATA_REGION_SIZE>; /* 4 KB */
+			no-map;
+		};
+	};
+
 	signature {
 		key-boot_key {
 			required = "conf";
@@ -188,3 +197,8 @@
 		};
 	};
 };
+
+&ddr_errata {
+	memory-region = <&ddr_errata_reserved>;
+	status = "okay";
+};
-- 
2.17.1

